fastlane/swift/ConfigItem.swift

```diff
@@ -0,0 +1,129 @@
+// ConfigItem.swift
+// Copyright (c) 2021 FastlaneTools
+
+//
+//  ** NOTE **
+//  This file is provided by fastlane and WILL be overwritten in future updates
+//  If you want to add extra functionality to this project, create a new file in a
+//  new group so that it won't be marked for upgrade
+//
+
+import Foundation
+
+public enum ConfigItem<T> {
+    case fastlaneDefault(T)
+    case userDefined(T)
+    case `nil`
+
+    func asRubyArgument(name: String, type: RubyCommand.Argument.ArgType? = nil) -> RubyCommand.Argument? {
+        if case let .userDefined(value) = self {
+            return RubyCommand.Argument(name: name, value: value, type: type)
+        }
```

Comment:
Why `.fastlaneDefault(T)` is ignored here? Is that for document purpose?

Reply:
We don't generate an argument when the case is not `.userDefined`, so `nil` will be returned and cleaned up in the `compactMap { $0 }` call before being sent into the socket.